### DISCUSSION TITLE
Top languages and private Count

### DISCUSSION THREAD

**Discussion by zentryxx** [anuraghazra/github-readme-stats#814] [Q&A] [answered]
Are private projects also counted? If not, there is something to add to be included. (Top Languages)

LG

**Comment by qwerty541** [comment, marked as answer]
There is no way we could see your personal repositories data. We can only see what GitHub provides.

Although you can try to self host with your own personal access token that might work:
https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance

**Reply by arribass** [reply]
I have deploy the repo with vercel and still can't see the top languages of my private repos? Is there any way way this can be done?
For further information this is my Readme
```
[![Arribass's GitHub stats](https://github-readme-stats.vercel.app/api?username=arribass&count_private=true)](https://github.com/arribass/github-readme-stats)

[![Arribass Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=arribass)](https://github.com/arribass/github-readme-stats)
```

**Reply by Qubik65536** [reply]
> I have deploy the repo with vercel and still can't see the top languages of my private repos? Is there any way way this can be done?
> For further information this is my Readme
> 
> ```
> [![Arribass's GitHub stats](https://github-readme-stats.vercel.app/api?username=arribass&count_private=true)](https://github.com/arribass/github-readme-stats)
> 
> [![Arribass Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=arribass)](https://github.com/arribass/github-readme-stats)
> ```

When you deploy your fork with Vercel, you'll get a new URL from it, and you should replace `github-readme-stats.vercel.app` with the URL provided by Vercel. And you should also add an environment variable `PAT_1` (the value is your GitHub Access Token with **repo** scope selected) to Vercel deployment if you haven't done it.

**Comment by Hephaest** [comment]
Hi @zentryxx @arribass, 

I have implemented this feature. There are only 4 steps to achieve this goal:

- Step 1: Fork my [repo](https://github.com/Hephaest/github-readme-stats).
- Step 2: Generate the **Github Personal Access Token** and select the **repo** scope.
- Step 3: Deploy the project on your own Vercel instance, and please don't forget to set the environment parameter `PAT_1=Your-Access-Token`. Two useful links you can follow:
  - [Github Guide](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance)
  - [Youtube Guide](https://www.youtube.com/watch?v=n6d4KHSKqGk&t=108s)
 - Step 4: Thanks for @Ayagikei 's reminder, please append the param `&count_private=true` to the URL.
   - For example: `https://YOUR-DOMAIN-NAME/api/top-langs/?username=YOUR-NAME&count_private=true` 
 
Related PR #1199

**Reply by Ayagikei** [reply]
Well done! I also succeeded in following your guidance.

**By the way:** don't forget to add the param `&count_private=true` to the URL

**Reply by ioogle** [reply]
Hi Heaphaest,

Thanks for your guidance, and now the private count for top languages is showing up. While the stats card still cannot show up private counts. Do you know what's the problem?

**Reply by yinzixie** [reply]
For me, the top languages are not showing up correctly but the stats card works, both URLs add param `&count_private=true`.

**Reply by Hephaest** [reply]
Dear @ioogle, sorry for the late reply since I unsubscribed the notification of this feed. I only modify the file related to top languages. It was weird to hear that the stats card cannot work as its design. Have you forget to add `&count_private=true` to the URL param. If not, I strongly recommend you create an issue and ask anuraghazra for help.

**Reply by Hephaest** [reply]
同学你好，你说的问题原因是你没有 fork 我的[版本](https://github.com/Hephaest/github-readme-stats)，需要按照我上述说的流程走。如果你直接 fork 原仓库没有用哈，因为我的 PR 还没有合入，而且原仓库没有实现该功能，所以自然就没有效果。

**Reply by yinzixie** [reply]
啊，确实，疏忽了。感谢~。

**Reply by yinzixie** [reply]
刚重试了一下，好像还是不行？你看看我哪里搞错了？感谢。

**Reply by Qubik65536** [reply]
> 刚重试了一下，好像还是不行？你看看我哪里搞错了？感谢。

@yinzixie 在fork之后，你还需要将这个fork过的仓库部署到vercel并且在部署过程中添加环境变量`PAT_1`，值是你的GitHub Access Token，之后，vercel会将你的repo部署到一个单独的域名，例如`readme-stats.domain.tld`，你还需要将你的readme中所有的域名替换掉。最后就是数据更新需要时间，等待即可。

**Reply by Hephaest** [reply]
> 刚重试了一下，好像还是不行？你看看我哪里搞错了？感谢。

感谢 @Cubik65536 的提示, @yinzixie 创建的 Github Access Token 记得把 **repo** 复选框勾选上(如下图)，这样才有获取私有库的权限哦。

<img width="600" alt="Screen Shot 2021-08-30 at 8 48 19 PM" src="https://user-images.githubusercontent.com/37981444/131341664-d1ab606b-ae91-4cfe-9264-711e9434f194.png">

确实存在数据更新延缓的问题(缓存机制)，但你只要根据上述的步骤操作的话应该很快就能看到效果了。

**Reply by Qubik65536** [reply]
> 同学你好，你说的问题原因是你没有 fork 我的[版本](https://github.com/Hephaest/github-readme-stats)，需要按照我上述说的流程走。如果你直接 fork 原仓库没有用哈，因为我的 PR 还没有合入，而且原仓库没有实现该功能，所以自然就没有效果。

@yinzixie 另外，我这边是直接fork的原仓库（`anuraghazra/github-readme-stats`），在使用了上述部署方法之后是可以看到私人仓库被计算的，该功能可能已被实现

仅作提醒

**Reply by yinzixie** [reply]
@Cubik65536 @Hephaest  感谢提醒，今天早上刷新了一下，已经可以了，完成后还要等待一段时间等数据更新。

**Comment by rickstaa** [comment]
> Hi @zentryxx @arribass,
> 
> I have implemented this feature. There are only 4 steps to achieve this goal:
> 
> * Step 1: Fork my [repo](https://github.com/Hephaest/github-readme-stats).
> * Step 2: Generate the **Github Personal Access Token** and select the **repo** scope.
> * Step 3: Deploy the project on your own Vercel instance, and please don't forget to set the environment parameter `PAT_1=Your-Access-Token`. Two useful links you can follow:
>   
>   * [Github Guide](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance)
>   * [Youtube Guide](https://www.youtube.com/watch?v=n6d4KHSKqGk&t=108s)
> * Step 4: Thanks for @Ayagikei 's reminder, please append the param `&count_private=true` to the URL.
>   
>   * For example: `https://YOUR-DOMAIN-NAME/api/top-langs/?username=YOUR-NAME&count_private=true`
> 
> Related PR #1199

@Hephaest First of all, thanks for trying to solve this issue. As explained by @qwerty541 there is no way for us to see users' private data. I checked your PR but I don't think it will benefit the public GRS instance. GraphQL's [isPrivate](https://docs.github.com/en/graphql/reference/objects#repository) property is used to check if a repository is private. The public API does not access users' private repositories, so this property will always be `false`. For a self-deployed Vercel instance, it is not needed since all private contributions are included by default.
